src/IO/Resources/SoundsLoader.cs

```diff
@@ -298,6 +304,41 @@ private bool TryParseConfigLine(string line, out Tuple<int, string, bool> songDa
             return true;
         }
 
+        /// <summary>
+        ///     Returns true filename from name, ignoring case since UO isn't consistent with file case (necessary for *nix)
+        /// </summary>
+        /// <param name="name">The filename from the music Config.txt</param>
+        /// <returns>a string with the true case sensitive filename</returns>
+        private static string GetTrueFileName(string name)
+        {
+            // don't worry about subdirectories, we'll recursively search them all
+            string dir = Settings.GlobalSettings.UltimaOnlineDirectory + $"/Music";
+
+            // Enumerate all files in the directory, using the file name as a pattern
+            // This will list all case variants of the filename even on file systems that
+            // are case sensitive.
+            Regex  pattern = new Regex($"^{name}.mp3", RegexOptions.IgnoreCase);
+            string[] fileList = Directory.GetFiles(dir, "*.mp3", SearchOption.AllDirectories).Where(path => pattern.IsMatch(Path.GetFileName(path))).ToArray();
+
+            if (fileList.Any())
```

Comment:
just check for `fileList != null && fileList.Length != 0`.
Avoiding linq is a good practice in a game env.

Reply:
Makes sense.

Given this, I'll also try to use `.Find()` instead of `.Where()` and will revise. Then I can eliminate the Linq dependency